pkg/controller/migplan/migplan_controller.go

```diff
@@ -445,3 +481,69 @@ func (r ReconcileMigPlan) deleteImageRegistryDeploymentForClient(client k8sclien
 	}
 	return nil
 }
+
+func (r ReconcileMigPlan) ensureMigAnalytics(plan *migapi.MigPlan) error {
+	migAnalytics := &migapi.MigAnalyticList{}
+	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	for _, migAnalytic := range migAnalytics.Items {
+		if migAnalytic.Spec.AnalyzeExtendedPVCapacity {
+			if plan.Spec.Refresh {
+				migAnalytic.Spec.Refresh = true
+				err := r.Update(context.TODO(), &migAnalytic)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+	}
+
+	pvMigAnalytics := &migapi.MigAnalytic{}
+	pvMigAnalytics.GenerateName = plan.Name + "-"
+	pvMigAnalytics.Namespace = plan.Namespace
+	pvMigAnalytics.Spec.AnalyzeExtendedPVCapacity = true
+	pvMigAnalytics.Annotations = map[string]string{MigPlan: plan.Name, CreatedBy: plan.Name}
+	pvMigAnalytics.Labels = map[string]string{MigPlan: plan.Name, CreatedBy: plan.Name}
+	pvMigAnalytics.OwnerReferences = append(pvMigAnalytics.OwnerReferences, metav1.OwnerReference{
+		APIVersion: plan.APIVersion,
+		Kind:       plan.Kind,
+		Name:       plan.Name,
+		UID:        plan.UID,
+	})
+	pvMigAnalytics.Spec.MigPlanRef = &kapi.ObjectReference{Namespace: plan.Namespace, Name: plan.Name}
+	err = r.Create(context.TODO(), pvMigAnalytics)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r ReconcileMigPlan) waitForMigAnalyticsReady(plan *migapi.MigPlan) (*migapi.MigAnalytic, error) {
+	migAnalytics := &migapi.MigAnalyticList{}
+	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
+	if err != nil {
+		return nil, err
+	}
+	for _, migAnalytic := range migAnalytics.Items {
+		if migAnalytic.Spec.AnalyzeExtendedPVCapacity == true {
+			if migAnalytic.Status.IsReady() {
+				return &migAnalytic, nil
+			}
+			if time.Now().Sub(migAnalytic.CreationTimestamp.Time) > 3 * time.Minute {
+				plan.Status.SetCondition(migapi.Condition{
+					Type:               miganalytic.NotReady,
```

Comment:
I would prefer the condition to be defined in MigPlan instead of MigAnalytic keeping concerns separate. This condition should be called something like `IntelligentPVResizingDisabled` or something similar that conveys the actual user impact.

Reply:
+1, Good catch!